### PR TITLE
feat(DMS): kafka instance support prepaid

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  kafka instance support prepaid
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  kafka instance support prepaid
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dms/' TESTARGS='-run TestAccKafkaInstance_prePaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccKafkaInstance_prePaid -timeout 360m -parallel 4
=== RUN   TestAccKafkaInstance_prePaid
=== PAUSE TestAccKafkaInstance_prePaid
=== CONT  TestAccKafkaInstance_prePaid
--- PASS: TestAccKafkaInstance_prePaid (837.71s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       837.763s
```

### CODE REVIEW

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 1]
```diff
@@ -637,6 +693,69 @@ func createKafkaInstanceWithProductID(ctx context.Context, d *schema.ResourceDat
...
+		return err
+	}
+	_, err = common.WaitOrderResourceComplete(ctx, bssClient, orderId, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return fmt.Errorf("error waiting for Kafka order resource %s complete: %s", orderId, err)
+	}
+	return nil
+}
+
+func waitForInstanceCreated(ctx context.Context, d *schema.ResourceData, client *golangsdk.ServiceClient, instanceID string) (string, error) {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: how about rename this function to `getInstanceOrderId` ?

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 2]
```diff
@@ -637,6 +693,69 @@ func createKafkaInstanceWithProductID(ctx context.Context, d *schema.ResourceDat
...
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"EMPTY"},
+		Target:       []string{"CREATING"},
+		Refresh:      kafkaInstanceCreatingFunc(client, instanceID),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		PollInterval: 500 * time.Millisecond,
```
Comment (ShiChangkuo) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go:732]: delay is missing

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 3]
```diff
@@ -288,6 +290,10 @@ func ResourceDmsKafkaInstance() *schema.Resource {
 					},
 				},
 			},
+			"charging_mode": common.SchemaChargingMode([]string{}),
+			"period_unit":   common.SchemaPeriodUnit([]string{}),
+			"period":        common.SchemaPeriod([]string{}),
```
Comment (ShiChangkuo) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]:
```suggestion
			"charging_mode": common.SchemaChargingMode(nil),
			"period_unit":   common.SchemaPeriodUnit(nil),
			"period":        common.SchemaPeriod(nil),
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 4]
```diff
@@ -288,6 +290,10 @@ func ResourceDmsKafkaInstance() *schema.Resource {
...
+			"auto_renew":    common.SchemaAutoRenewUpdatable(nil),
```
Comment (ShiChangkuo) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go:296]: `auto_new` can be updated, but it's not used in Update method.

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: it's a good advice, done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go:732]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go:296]: this param can not be updated in this resource, so it has been marked as ForceNew

**Pull request merged**